16-google/cms.ts

```diff
@@ -0,0 +1,28 @@
+import { Event, normal, metronome, TimedDependency, Timeout } from "../../src";
+
+export class ClusterManagementSoftware extends TimedDependency {
+  public queue1: number = 0;
+  public queue2: number = 0;
+  private queueLimit: number = 2500;
+  constructor(protected timeouts: Timeout[]) {
+    super();
+  }
+
+  async workOn(event: Event): Promise<void> {
```

Comment:
I'm a little confused at what this stage is for.

I see it randomly decides which stage to send the event on to, and it keeps track of how often it is sending events to each stage (and capping that at some value) but what part of the incident report does this behavior capture? Is there somewhere where it says that there is a max rate it sends events so some stages? If so, you can cap this value in the stage by assigning a queue with a set number of workers instead of having to manage this outside of the stage. Or you can monitor the concurrent events currently being processed in a different stage from this stage, without having to keep track here.